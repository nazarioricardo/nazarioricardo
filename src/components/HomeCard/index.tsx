import { Download } from "@mui/icons-material";
import LinkButton from "@nr/components/LinkButton";
import Link from "next/link";
import styles from "./styles.module.css";
import TitleCard from "../TitleCard";
import Socials from "../Socials";

function HomeCard() {
  return (
    <>
      <TitleCard className={styles.homeCard}>
        <section>
          <div className={styles.title}>
            <h1 className={styles.name}>Ricardo Nazario</h1>
            <p className={styles.email}>nazarioricardo@gmail.com</p>
            <Socials />
          </div>

          <p className={styles.bio}>
            {`Hello. I'm a software developer who loves creating joyful products
  that make life a little nicer for everyone.`}
          </p>
          <br />
          <p className={styles.bio}>
            {`I've worked as a full stack and mobile developer for the past 6
  years, making products in the music, medical, and energy
  industries.`}
          </p>
        </section>
        <section className={styles.bottomContent}>
          <LinkButton
            text="View C.V."
            href="/RicardoNazario_Resume.pdf"
            Icon={Download}
          />
        </section>
      </TitleCard>
      <div className={styles.zFill} />
    </>
  );
}

export default HomeCard;
