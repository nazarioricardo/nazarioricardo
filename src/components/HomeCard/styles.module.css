.homeCard {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  min-height: 100vh;
  min-width: 40vw;
  max-width: 40vw;
  border-top-left-radius: 0 !important;
  border-bottom-left-radius: 0 !important;
  position: fixed !important;
  top: 0 !important;
  z-index: 9;
}

.zFill {
  min-height: 100vh;
  min-width: 40vw;
  max-width: 40vw;
}

.title {
  padding-bottom: 1rem;
}

.name {
  font-size: 3rem;
  font-family: var(--font-jomolhari), "Hoeffler Text", serif;
  font-weight: 400;
}

.bottomContent {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  padding-top: 24px;
  gap: 24px;
  font-size: 0.8rem;
}

.githubLink,
.link {
  text-decoration: underline;
  font-weight: bold;
}

.githubLink:hover,
.link:hover {
  opacity: 0.5;
}

@media only screen and (max-width: 1360px) {
  .homeCard {
    padding-left: 4%;
    padding-right: 4%;
  }
}

@media only screen and (max-width: 1170px) {
  .homeCard {
    min-width: 100%;
    max-width: 100%;
    min-height: auto;
    height: auto;
    border-radius: 50px !important;
    z-index: 0;
    animation: fadeIn 0.5s;
    position: relative !important;
    top: 0 !important;
  }

  @keyframes fadeIn {
    0% {
      opacity: 0;
    }

    100% {
      opacity: 1;
    }
  }

  .header {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    min-height: 48px;
    background-color: transparent;
    cursor: pointer;
    font-size: 1rem;
    color: rgb(var(--foreground-rgb));
  }

  .link {
    min-height: 52px;
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    align-items: center;
    gap: 6px;
  }

  .title > h1 {
    font-size: 2rem;
  }

  .zFill {
    display: none;
  }
}
