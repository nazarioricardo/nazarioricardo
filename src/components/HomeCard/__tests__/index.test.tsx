import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import HomeCard from "../index"; // adjust this import according to your file structure

describe("HomeCard", () => {
  test("renders HomeCard component", () => {
    render(<HomeCard />);
    expect(screen.getByText("Ricardo Nazario")).toBeInTheDocument();
    expect(screen.getByText("nazarioricardo@gmail.com")).toBeInTheDocument();
    expect(
      screen.getByText(
        /Hello. I'm a software developer who loves creating joyful products/
      )
    ).toBeInTheDocument();
    expect(
      screen.getByText(
        /I've worked as a full stack and mobile developer for the past 6/
      )
    ).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<HomeCard />);
    expect(asFragment()).toMatchSnapshot();
  });

  // Add more tests as needed
});
