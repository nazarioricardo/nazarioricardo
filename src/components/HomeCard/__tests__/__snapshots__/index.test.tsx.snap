// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeCard matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="titleCard homeCard"
  >
    <div>
      <div
        class="title"
      >
        <h1
          class="name"
        >
          Ricardo Nazario
        </h1>
        <p
          class="email"
        >
          nazarioricardo@gmail.com
        </p>
      </div>
      <p
        class="bio"
      >
        Hello. I'm a software developer who loves creating joyful products
  that make life a little nicer for everyone.
      </p>
      <br />
      <p
        class="bio"
      >
        I've worked as a full stack and mobile developer for the past 6
  years, making products in the music, medical, and energy
  industries.
      </p>
    </div>
    <div
      class="bottomContent"
    >
      <div>
        <p>
          <b>
            Dear Visitor,
          </b>
        </p>
        <br />
        <p>
          This page launched December 9, 2023. There are a few things I still need to add.
        </p>
        <br />
        <p>
          If you're curious, check out the current 
          <a
            class="githubLink"
            href="https://github.com/nazarioricardo/nazarioricardo/issues"
            target="_blank"
          >
            github issues
          </a>
          .
        </p>
      </div>
      <a
        class="linkButton undefined"
        href="/Ricardo Nazario Resume.pdf"
      >
        View C.V. 
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="DownloadIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M5 20h14v-2H5v2zM19 9h-4V3H9v6H5l7 7 7-7z"
          />
        </svg>
      </a>
    </div>
  </section>
</DocumentFragment>
`;
